simconvoi.h

```diff
@@ -495,7 +496,23 @@ class convoi_t : public sync_steppable, public overtaker_t
 	/// Pushes the convoy stopping time at the current halt to the schedule
 	void push_convoy_stopping_time();
 
+	// Reverses the order of the convoy.
+	// @author: jamespetts
+	uint8 vehicle_count;
+
+
+	bool reverse_parent_children;// Reverse the order of the coupling/coupled convois 
 public:
+
+	bool reversing;// Whether this convoy's vehicles are currently arranged in reverse order. The flag for calculation of position.
+	bool is_reversed() const;
+	// Reorder the vehicle array
+	// Can be executed even with a vehicle array that does not belong to convoy for UI
+	void reverse_order(bool rev);
+	void reversing_immediately(bool rev);
+	static void execute_reverse_order(array_tpl<vehicle_t*> &vehicles, uint8 vehicle_count);
+	// Reverse the order of the coupling/coupled convois
+	static void execute_reverse_parent_children(convoihandle_t self);
```

Comment:
- [naming] Isn't just `reverse_convoy_coupling` enough?
- This can be a member function.

```suggestion
	static void reverse_convoy_coupling(convoihandle_t self);
```

Reply:
execute_reverse_couplingは編成反転の実行関数です
実行のための場合分けや事後処理を区別したためこのようになっています

Reply:
こちらは、現状場合分けや例外処理が発生していないので間の関数を入れていなかったですが、必要でしょうか？

Reply:
関数は一つで良いと思います。
`execute` という接頭語にあまり意味がないのでいらないのではないかと思ったのと、対象を`convoihandle_t` で受け取っているのでそれならば初めからメンバ関数にした方が良いのではと考えた次第です。